core/pom.xml

```diff
@@ -9,6 +10,71 @@
     <artifactId>smallrye-context-propagation</artifactId>
     <name>smallrye-context-propagation</name>
 
+    <properties>
+        <java9.sourceDirectory>${project.basedir}/src/main/java9</java9.sourceDirectory>
+        <java9.build.outputDirectory>${project.build.directory}/classes-java9</java9.build.outputDirectory>
+    </properties>
+
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>compile-java9</id>
+                        <phase>compile</phase>
+                        <configuration>
+                            <tasks>
+                                <property name="compile_classpath" refid="maven.compile.classpath" />
+                                <mkdir dir="${java9.build.outputDirectory}" />
+                                <javac fork="true" executable="${env.JAVA_HOME}/bin/javac" srcdir="${java9.sourceDirectory}"
```

Comment:
Also relevant to my previous comment, assuming we will enforce building on JDK 9+ (most likely 11), then you shouldn't need this. Having to have env variable defined to make a build pass is sort of weird, ideally it should all "just work" with `mvn clean install` ;-)